File: Maximize Happiness of Selected Children.py
# https://leetcode.com/problems/maximize-happiness-of-selected-children/?envType=daily-question&envId=2024-05-09
# Maximize Happiness of Selected Children

class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
    
        total_happiness = 0
        for i in range(k):
            decreased_value = happiness[i] - i
            total_happiness += max(0, decreased_value)
        
        return total_happiness

File: Maximize Happiness of Selected Children.js
// https://leetcode.com/problems/maximize-happiness-of-selected-children/?envType=daily-question&envId=2025-12-25
// Maximize Happiness of Selected Children

/**
 * @param {number[]} happiness
 * @param {number} k
 * @return {number}
 */
var maximumHappinessSum = function(happiness, k) {
    happiness.sort((a, b) => b - a);

    let total = 0;

    for (let i = 0; i < k; i++) {
        const gain = happiness[i] - i;
        if (gain <= 0) break;
        total += gain;
    }

    return total;
};